hw/ip/i2c/doc/theory_of_operation.md

```diff
@@ -380,7 +380,7 @@ Each sub-sequence has a terminal event--generically labeled "[completed]" which
 
 However, all transitions which are dependent on formatting flags are shown explicitly in this figure.
 
-![](../doc/I2C_state_diagram.svg)
+![](../doc/i2c_controller_states.svg)
```

Comment:
Nit. This state diagram can be made with mermaid, which is supported by github.
```mermaid
stateDiagram-v2
    [*] --> Idle
    Idle --> [*]
    Idle --> Active
    Active --> Write
    Write --> Stop
    Stop --> Idle
```

Reply:
If I have time, I'll do that in a follow-up PR today. It looks like going from graphviz to mermaid should be pretty straightforward.